doc/source/tune/_tutorials/tune-sklearn.py

```diff
@@ -0,0 +1,158 @@
+
+Run ``pip install ray[tune] tune-sklearn`` to get started.
+
+Walkthrough
+-----------
+
+Let's compare Tune's Scikit-Learn APIs to the standard scikit-learn GridSearchCV. For this example, we'll be using ``TuneGridSearchCV`` with a `SGDClassifier`_.
+
+.. _`digits dataset`: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_digits.html
+.. _`SGDClassifier`: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html
+
+To start out, change the import statement to get tune-scikit-learn’s grid search cross validation interface:
+
+"""
+# from sklearn.model_selection import GridSearchCV
+from ray.tune.sklearn import TuneGridSearchCV
+
+#######################################################################
+# And from there, we would proceed just like how we would in Scikit-Learn’s interface!
+#
+# The `SGDClassifier`_ has a ``partial_fit`` API, which enables it to stop fitting to the data for a certain hyperparameter configuration.
+# If the estimator does not support early stopping, we would fall back to a parallel grid search.
+
+# Other imports
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import SGDClassifier
+from sklearn.datasets import make_classification
+import numpy as np
+
+# Create dataset
+X, y = make_classification(
+    n_samples=11000,
+    n_features=1000,
+    n_informative=50,
+    n_redundant=0,
+    n_classes=10,
+    class_sep=2.5)
+x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=1000)
+
+# Example parameters to tune from SGDClassifier
+parameter_grid = {"alpha": [1e-4, 1e-1, 1], "epsilon": [0.01, 0.1]}
+
+#######################################################################
+# As you can see, the setup here is exactly how you would do it for Scikit-Learn. Now, let's try fitting a model.
+
+tune_search = TuneGridSearchCV(
+    SGDClassifier(),
+    parameter_grid,
+    early_stopping=True,
+    max_iters=10)
+
+import time  # Just to compare fit times
+start = time.time()
+tune_search.fit(x_train, y_train)
+end = time.time()
+print("Tune Fit Time:", end - start)
+
+#######################################################################
+# Note the slight differences we introduced above:
+#
+#  * a `early_stopping`, and
+#  * a specification of `max_iters` parameter
+#
+# The ``early_stopping`` parameter allows us to terminate unpromising configurations. If ``early_stopping=True``,
+# TuneGridSearchCV will default to using Tune's ASHAScheduler. You can pass in a custom
+# algorithm - see :ref:`Tune's documentation on schedulers <tune-schedulers>` here for a full list to choose from.
+# ``max_iters`` is the maximum number of iterations a given hyperparameter set could run for; it may run for fewer iterations if it is early stopped.
+#
+# Try running this compared to the GridSearchCV equivalent, and see the speedup for yourself!
+
```

Comment:
Should we add an example output for the search times here? Since this is non-interactive, this might help people that are just reading through the tutorial to get a sense for the speedup. Of course this would need a disclaimer that training times differ for different machines.

Reply:
On my personal machine (dual-core i5 8 GB RAM), Tune ran for ~33 seconds and sklearn ran for ~86 seconds.